Project/VetAppointmentAPI/VetAppointmentAPI/VetAppointment.Infrastructure/Repositories/Base/BaseRepository.cs

```diff
@@ -16,6 +16,7 @@ protected BaseRepository(IDatabaseContext context)
         public virtual async Task<T> Add(T entity)
         {
             await context.Set<T>().AddAsync(entity);
+            await context.Save();
```

Comment:
Well we either add a context.Save() to all data manipulation methods (Update, Delete, Add) or to none, cuz it's not intuitive otherwise.